plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java

```diff
@@ -342,15 +344,22 @@ private Block getArrayBlock(int rowIndex, int columnIndex)
     private Slice getSlice(int rowIndex, int columnIndex)
     {
         Type trinoType = getType(columnIndex);
+        DataTable dataTable = currentDataTable.getDataTable();
+
         if (trinoType instanceof VarcharType) {
-            String field = currentDataTable.getDataTable().getString(rowIndex, columnIndex);
+            String field = dataTable.getString(rowIndex, columnIndex);
             return getUtf8Slice(field);
         }
         if (trinoType instanceof VarbinaryType) {
-            return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getString(rowIndex, columnIndex)));
+            if (dataTable instanceof DataTableImplV4) {
+                return Slices.wrappedBuffer(dataTable.getBytes(rowIndex, columnIndex).getBytes());
+            }
+            else {
+                return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getString(rowIndex, columnIndex)));
+            }
```

Comment:
Could you please add a test?

Reply:
Make dataTable to v4 for half of the tests.

Reply:
All tests passed without the `pinot-server.conf` change. I suppose it's not covered by the existing tests.

Reply:
Ah, it's covered in the test using `TestPinotConnectorSmokeTest`, `TestPinotLatestConnectorSmokeTest`, `TestPinotLatestNoGrpcConnectorSmokeTest`, since the are using `pinot-server.conf`.

Once added the config `pinot.server.instance.currentDataTableVersion=4`, Pinot server will start returning v4 DataTable, before it's using v3 for data table(Ref: https://github.com/apache/pinot/pull/10543/files#diff-2573700946371cb7a52aaf7d71bb966cfa8e2704f28c3fdd11254d5dd7176bf9L37).